Remake PR: Uninstall CTT PowerShell Profile

Cryostrixx opened the pull request:
## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [x] UI/UX improvement

## Description

Implements the ability to uninstall the CTT PowerShell Profile and all its related dependencies.

## Testing

I have tested the profile uninstallation functionality by running it in the following test scenarios:

- Baseline: Installing the CTT PowerShell Profile
- Test 1: Uninstallation of OhMyPosh dependency: [✅] Successful
- Test 2: Uninstallation of Nerd Fonts dependency: [❓] Conditional
    - Additional Context: See `Notes for Test 2` in `Additional Information`.
- Test 3: Uninstallation of Terminal-Icons dependency: [✅] Successful
- Test 4: Uninstallation of Zoxide dependency: [✅] Successful
- Test 5: Skipping uninstallation of in-use dependencies: [✅] Successful
- Test 6: Uninstallation of CTT PowerShell Profile: [✅] Successful

## Impact

- Covered by `## Description`, `## Testing`, and `## Additional Information`.

## Issue related to PR

Resolves #2836

## Additional Information

### New Files Created

- `$PSProfile.hash`: A file that stores the hash of the CTT PowerShell Profile; this is used to check if the profile is installed on the system.
  - Additional Context:
    - The path is derived from the `$PSProfile` variable with `.hash` being appended to the end of the file path.
    - Example Hash File Path: `C:\Users\<user>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1.hash`.
    - This file will be cleaned up and removed when the CTT PowerShell Profile is uninstalled from the system.
    - `$PSProfile.bak`: A file that is used as a backup of the PowerShell profile prior to the CTT PowerShell Profile being installed on the system.
      - Additional Context:
        - The path is derived from the `$PSProfile` variable with `.bak` being appended to the end of the file path.
        - Example Backup File Path: `C:\Users\<user>\Documents\PowerShell\Microsoft.PowerShell_profile.ps1.bak`.
        - This file will be moved back to the original PowerShell profile's location when the CTT PowerShell Profile is uninstalled from the system.

### Notes For Test 2

- Note 1: Nerd Fonts won't always be installed due to an issue with the setup script for the PowerShell profile.
- Note 2: If the font is not installed, the user will see the message: `===> Already Uninstalled: Nerd Fonts. <===`.

### Possible Change

I'm not sure if the documentation files in `./docs/dev/features/PowerShell-Profile` are necessary, I added them for completeness but if they aren't needed please let me know so I can remove them.

## Checklist

- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

ChrisTitusTech reviewed (approved):
Thanks for the uninstall addition and the fix for updating powershell on setup.

Cryostrixx commented:
> Thanks for the uninstall addition and the fix for updating powershell on setup.

First off: Thanks for merging the new feature, and please let me know if any issues come up with it.
Secondly: I did check and this appears to be a known issue, but I want to query about it regardless:
Just to get your input: Should I remove the documentation files I added through a new PR? I noticed most of the documentation links do not work at the moment, presumably because they are being refactored. Here is a list of broken links, for reference:
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Powershell7.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/RestorePoint.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/EdgeDebloat.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/DeleteTempFiles.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/AH.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/ConsumerFeatures.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/DVR.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Hiber.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Home.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Loc.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Powershell7Tele.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Storage.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Tele.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Wifi.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/EndTaskOnTaskbar.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/IPv46.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/DiskCleanup.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/LaptopHibernation.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Essential-Tweaks/Services.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Shortcuts/Shortcut.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/DebloatAdobe.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/BlockAdobeNet.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/DisableBGapps.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/DisableFSO.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/DisableLMS1.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/Disableipsix.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/RemoveCopilot.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/DisableNotifications.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/Teredo.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/changedns.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/DeBloat.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/RemoveHomeGallery.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/RemoveEdge.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/RemoveOnedrive.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/OOSUbutton.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/RightClickMenu.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/Display.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/z--Advanced-Tweaks---CAUTION/UTC.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/BingSearch.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/TaskbarAlignment.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/DarkMode.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/DetailedBSoD.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/MouseAcceleration.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/NumLock.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/TaskbarSearch.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/ShowExt.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/HiddenFiles.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/SnapFlyout.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/SnapSuggestion.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/SnapWindow.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/StickyKeys.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/TaskView.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/VerboseLogon.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Customize-Preferences/TaskbarWidgets.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Performance-Plans/AddUltPerf.md
* https://christitustech.github.io/winutil/docs/dev/tweaks/Performance-Plans/RemoveUltPerf.md
* https://christitustech.github.io/winutil/docs/dev/features/Fixes/RunAdobeCCCleanerTool.md
* https://christitustech.github.io/winutil/docs/dev/features/Fixes/Network.md
* https://christitustech.github.io/winutil/docs/dev/features/Fixes/Update.md
* https://christitustech.github.io/winutil/docs/dev/features/Fixes/Autologin.md
* https://christitustech.github.io/winutil/docs/dev/features/Fixes/DISM.md
* https://christitustech.github.io/winutil/docs/dev/features/Fixes/Winget.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/control.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/network.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/power.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/printer.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/region.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/sound.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/system.md
* https://christitustech.github.io/winutil/docs/dev/features/Legacy-Windows-Panels/user.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/dotnet.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/DisableLegacyRecovery.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/DisableSearchSuggestions.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/RegBackup.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/EnableLegacyRecovery.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/EnableSearchSuggestions.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/hyperv.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/Install.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/legacymedia.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/nfs.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/Sandbox.md
* https://christitustech.github.io/winutil/docs/dev/features/Features/wsl.md

-----

Apologies in advance for derailing this a bit, I just wanted to confirm it as the links I added don't work either.
